src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -93,9 +98,7 @@ public function loadChoiceList($value = null)
             ? $this->objectLoader->getEntities()
             : $this->manager->getRepository($this->class)->findAll();
 
-        $this->choiceList = $this->factory->createListFromChoices($objects, $value);
-
-        return $this->choiceList;
+        return $this->choiceList = new ArrayChoiceList($objects, $value);
```

Comment:
When the `ChoiceListFactory` create first a `LazyChoiceList` instance (the first and mandatory call) all the decorators get pass the `$value` so they can resolve it.

Once this step done the `LazyChoiceList` will be waken up to use its loader and resolved value all together.

The thing is actually instead of return an `ArrayChoiceList` instance with the loaded choices, the `DoctrineChoiceLoader` (which is the only `ChoiceLoaderInterface` implemented in core) get passed the hole factory with all the decorators (and might add some in the future as it is one a the few BC ways to add feature for now).

I think this final call should be delegated, that remains coherent with the interfaces and has a performance impact.

Anyway it does not break the decorator resolution.